Brubeck broker and streamr-client examples (#9)
streamr-dev/examples: BrokerApiNode/README.md

# Broker APIs

> This example demostrates how to use the Streamr Broker API's from NodeJS. Node version 14/16 or higher is recommended.

Reviewer: ```suggestion
This example demostrates how to use the Streamr Broker API's from NodeJS. Node version 16 is recommended.
```

## Broker & SDKs

> - [Examples directory](https://github.com/streamr-dev/examples/blob/ECODR-34-broker-and-client/NodeJs/src/broker/)

Reviewer: These links will probs break when this gets merged? Can you do a find + replace for `blob/ECODR-34-broker-and-client/`

> The Broker acts as a relayer node from it's HTTP/WS/MQTT APIs into the Streamr Network

Reviewer: ```suggestion
The Broker node ships with interface plugins, which can be used to publish and subscribe to data from applications over off-the-shelf protocols HTTP, Websocket, and MQTT.

The plugins expose ports and API endpoints which can be used to publish and subscribe to data using the identity of the Broker node.
```

### Broker setup
```shell
$ npm install streamr-broker --global

> $ npm run broker:start

Reviewer: I believe these instructions install globally, but try to run the package locally? 

I think it should be, 
$ npm install streamr-broker --global
$ streamr-broker-init
$ streamr-broker .streamr/broker-config.json

```
The [default config](https://github.com/streamr-dev/examples/blob/ECODR-34-broker-and-client/NodeJs/broker-config.json) has the HTTP, WS and MQTT plugins enabled by default to allow for SDK interaction.

You can also start the broker with the apiKey `0x1c4343df92f5370208232782c373fa691c3543bdf4c40adfd406c87103b18fc2` by running:

```
$ npm run broker:start-api-key
```

In this setup you'll have to provide one of the registered apiKeys under `src/config-api-key.json` to be able to interact with the broker node. 

### HTTP Endpoint

> - [API Reference](https://api-explorer.streamr.com/)

Reviewer: The API explorer is for the mostly deprecated `core-api`. Is there anything that you reach for from the explorer?

One can exercise the [`publish`](https://github.com/streamr-dev/examples/blob/ECODR-34-broker-and-client/NodeJs/src/broker/http-publish.js) using the HTTP plugin on the broker. 

> In order to run the example execute from your terminal:

Reviewer: ```suggestion
In order to run this repo's HTTP publish example, execute from your terminal:
```

```shell
$ npm run http:publish
```

> If you are running your broker config with apiKeys you'll need to run:

Reviewer: ```suggestion
If you are running your broker config with `apiKeys`, you'll need to run:
```

```shell
$ npm run http:api-key-publish
```

### WS Endpoint
> Lacks apiKey examples for authorized brokers

One can exercise the [`publish`](https://github.com/streamr-dev/examples/blob/ECODR-34-broker-and-client/NodeJs/src/broker/ws-publish.js) and [`subscribe`](https://github.com/streamr-dev/examples/blob/ECODR-34-broker-and-client/NodeJs/src/broker/ws-subscribe.js) using the WebSockets plugin on the broker. 

In order to run the respective examples execute from your terminal:
```shell
$ npm run ws:subscribe
$ npm run ws:publish
```

If you are running your broker config with apiKeys you'll need to run:
```shell
$ npm run ws:api-key-subscribe
$ npm run ws:api-key-publish
```

**Things to remember**

> - Any ws client library can be used to interact with the broker

Reviewer: ```suggestion
- Any WebSocket client library can be used to interact with the broker
```
